### PR TITLE
doc: add mds state transition diagram

### PR DESCRIPTION
**Pull request by batrick**
Fixes: http://tracker.ceph.com/issues/22989

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
Looks nice. Does that file structure actually link it directly into the docs?

**Comment by batrick** [comment]
Not yet. I also forgot to mention the legend:

double lined shape: rank is "in"
hexagon: no MDS holding rank/state

**Comment by batrick** [comment]
render docs please

**Comment by batrick** [comment]
jenkins render docs

**Comment by batrick** [comment]
jenkins render docs

**Comment by batrick** [comment]
jenkins render docs

**Comment by batrick** [comment]
jenkins retest

**Comment by batrick** [comment]
jenkins render docs

**Comment by batrick** [comment]
jenkins render docs

**Comment by batrick** [comment]
jenkins render docs

File: doc/cephfs/mds-states.rst [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+recovering is journal and other metadata.
+
+
+::
+
+    up:resolve
+
+The MDS enters this state from ``up:replay`` if the Ceph file system has
+multiple ranks (including this one), i.e. it's not a single active MDS cluster.
+The MDS is trying to learn the authorities for subtrees in the file system.
```
Comment (ukernel) [inline doc/cephfs/mds-states.rst]: I think 'resolve' stands for 'resolve in-doubt transaction of  two phrase commit"

Comment (batrick) [reply doc/cephfs/mds-states.rst]: You're quite right. Thanks! I've fixed it.

File: doc/cephfs/mds-states.rst [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+::
+
+    up:standby_replay
+
+The MDS is following the journal of another ``up:active`` MDS. Should the
+active MDS fail, having a standby MDS in replay mode is desirable as the MDS is
+replaying the live journal and will more quickly takeover. A downside to having
+standby replay MDSs is that they are not available to takeover for any other
+MDS that fails, only the rank, fscid, or named MDS they follow.
```
Comment (gregsfortytwo) [inline doc/cephfs/mds-states.rst]: I'm not sure specifying the "rank, fscid, name" options for standby-replay actually helps here. Maybe drop that claus down to just "the MDS they follow".

File: doc/cephfs/mds-states.rst [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+metadata journal, and, if the last rank (0), evict all clients and shutdown
+(see also :doc:`administration/`).
+
+
+::
+
+    up:replay
+
+The MDS taking over a failed rank. This state represents that the MDS is
+recovering is journal and other metadata.
```
Comment (gregsfortytwo) [inline doc/cephfs/mds-states.rst]: recovering **its** journal

**Comment by batrick** [comment]
@gregsfortytwo done thanks!

File: doc/cephfs/mds-states.rst [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+This is the normal operating state of the MDS. It indicates that the MDS
+and its rank in the file system is available.
+
+
+::
+
+    up:standby
+
+The MDS is available to takeover for a failed rank (see also :doc:`standby/`).
```
Comment (alfredodeza) [inline doc/cephfs/mds-states.rst]: Please don't use paths to link to other documents, use the `ref` syntax (https://github.com/ceph/ceph/blob/master/doc/start/documenting-ceph.rst#toc-and-hyperlinks )

File: doc/cephfs/mds-states.rst [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+::
+
+    up:stopping
+
+When a rank is deactivated (stopped), the monitors command an active MDS to
+enter the ``up:stopping`` state. In this state, the MDS accepts no new client
+connections, migrates all subtrees to other ranks in the file system, flush its
+metadata journal, and, if the last rank (0), evict all clients and shutdown
+(see also :doc:`administration/`).
```
Comment (alfredodeza) [inline doc/cephfs/mds-states.rst]: same here, please use the `ref` directive

File: doc/cephfs/mds-states.rst [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+    $ ceph fs dump
+    ...
+    max_mds 1
+    in      0
+    up      {}
+    failed  
+    damaged 0
+    ...
+
+Rank 0 has become damaged (see also :doc:`disaster-recovery/`) and placed in
```
Comment (alfredodeza) [inline doc/cephfs/mds-states.rst]: ditto about `ref` directive

File: doc/cephfs/mds-states.rst [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+    ...
+    max_mds 1
+    in      0
+    up      {}
+    failed  
+    damaged 
+    stopped 1
+    ...
+
+The rank has been stopped by reducing ``max_mds`` (see also :doc:`multimds/`).
```
Comment (alfredodeza) [inline doc/cephfs/mds-states.rst]: ditto

File: doc/cephfs/mds-states.rst (hunk 4 above)
Comment (batrick) [reply doc/cephfs/mds-states.rst]: Fixed! Thanks for the link/review!

**Comment by batrick** [comment]
jenkins render docs

**Pull request merged**